xdsl/dialects/builtin.py

```diff
@@ -731,27 +734,27 @@ def verify(self):
 
     @staticmethod
     def create_dense_int_or_index(typ: IntegerType | IndexType,
-                                  data: List[int | IntAttr]) -> DenseArrayBase:
```

Comment:
Why do we remove that feature here?

Reply:
My impression was that it's not used and not helpful. The user is very unlikely to have a mix of `int`s and `IntAttr`s in a single list, so they either have a `list[int]` or `list[IntAttr]` in both cases, by using this constructor, there's an O(n) cost of dynamic checks for each element. The proposed change has two options, one for the `list[int]`, the other (`DenseArrayBase([typ, ArrayAttr(attr_list)])` directly) for `list[IntAttr]`.

Reply:
Feature is mostly back